internal/server/errors.go

```diff
@@ -77,6 +75,24 @@ func (a *API) sendAPIError(c *gin.Context, err error) {
 	c.Abort()
 }
 
+// trimErrorMsg attempts to prevent duplicate messages in error responses by
+// trimming the errBase message from errFull. If the two errors contain identical
+// messages then an empty string is returned. The errBase text is only removed
+// if it appears as a prefix or suffix to errFull.
+func trimErrorMsg(errBase error, errFull error) string {
+	base, full := errBase.Error(), errFull.Error()
+
+	switch {
+	case base == full:
+		return ""
+	case strings.HasPrefix(full, base+": "):
+		return strings.TrimPrefix(full, base+": ")
+	case strings.HasSuffix(full, ": "+base):
+		return strings.TrimSuffix(full, ": "+base)
+	}
```

Comment:
why do we want to trim the base error message out? Seems like it could only add to clarity. Keep in mind some bad clients don't show http status codes to users, but they do show the full message.

Reply:
I've reverted this part for now, and focused this PR on the client side changes.